### ISSUE TITLE
Grafana version 0.8.5 fails to deploy dashboards using dashboard-import-job on k8s v1.9.4

### ISSUE THREAD

**Issue by hmacias** [helm/charts#4708]
**Is this a BUG REPORT or FEATURE REQUEST?** BUG REPORT

**Version of Helm and Kubernetes**:

```
 ~  helm version
Client: &version.Version{SemVer:"v2.4.2", GitCommit:"82d8e9498d96535cc6787a6a9194a76161d29b4c", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
```

**Kubernetes version:** 1.9.4-gke.1

**Which chart**:

name: grafana
**version:** 0.8.5
**appVersion:** 5.0.4

**What happened**:

Grafana deployment fails when using *dashboardImports.dashboard* option to import dashboards on the fly when downloading JSON files. Job errors out with  **Read-only file system** message when wget tries to write JSON file to disk.

**What you expected to happen**:
Dashboard JSON files should be downloaded and passed to Grafana API for installation.

**How to reproduce it** (as minimally and precisely as possible):

1. Update **values.yaml** file to look like (extract):
```
dashboardImports:
  ## If true, a Job will be created to import dashboards
  ## Default: false
  ##
  enabled: true

  ## How long should it take to commit failure
  ## Default: 30
  ##
  activeDeadlineSeconds: 30

  ## Docker image to run import process
  ## Default: jgoclawski/wget:1.0
  ##
  image: jgoclawski/wget:1.0

  ## Grafana dashboard files to import as ConfigMap entries
  # files: {}

  ## Grafana dashboard URL to import (this should be a JSON file, not the HTML page)
  # Example:
  dashboards:
     Kubernetes_Cluster_Mon.json: https://grafana.com/api/dashboards/315/revisions/3/download
     NGinx_VTS_Stats.json: https://grafana.com/api/dashboards/2949/revisions/2/download
```
2. Run `helm upgrade -i monitor-dashboards ./ --namespace management --debug`

Job to download Dashboard JSON files fails with:

```
kubectl logs -n kube-system job/monitoring-grafana-import-dashboards
Found 3 pods, using pod/monitoring-grafana-import-dashboards-dkqf2
+ set -e
+ cd /var/lib/grafana/import
+ wget -O Kubernetes_Cluster_Mon.json https://grafana.com/api/dashboards/315/revisions/3/download
Kubernetes_Cluster_Mon.json: Read-only file system

```
**Anything else we need to know**:
Deployment succeeds if I don't try to import dashboards. Note that I got this to work by changing the bash script in **templates/dashboards-import-job-configmap.yaml** to use /tmp to download JSON files and dashboards were installed in Grafana correctly. Something like (extract of bash script, the rest looks exactly the same as in repo):

```
 init.sh: |
    #!/bin/sh
    set -e
    # cd /var/lib/grafana/import
    cd /tmp
    {{- range $name, $val := .Values.dashboardImports.dashboards }}
```

Fix should be simple: we can modify this script to copy any JSON files found at **/var/lib/grafana/import** to a **/tmp/grafana/import** dir and then proceed to download any other requested dashboards JSON files to finally pass them over to Grafana API for installation.

Thoughts?

**Comment by Jokero** [comment]
Also faced with the problem using minikube but can't find any specific logs.

values.yaml:
```yaml
server:
  image: "grafana/grafana:5.0.4"
  adminUser: admin
  adminPassword: admin

  persistentVolume:
    storageClass: standard
    size: 1Gi

  setDatasource:
    enabled: true
    datasource:
      name: prometheus
      type: prometheus
      url: http://prometheus:9090

  dashboardImports:
    enabled: true
    dashboards:
      cluster.json: https://grafana.com/api/dashboards/315/revisions/3/download
```
Data source is configured in grafana, but there is no any dashboard. The same thing when I use `dashboardImports.files`. Looks like import is broken.

**Comment by Jokero** [comment]
@bivas, @jgoclawski does import work for you? Maybe I'm doing something wrong?

**Comment by bivas** [comment]
I still haven't tried this version (`wget`). Still using the previous (`curl`) one I merged.

I know _that_ version works.

**Comment by jgoclawski** [comment]
The reason you see this bug, is an update in Kubernetes (1.7.14, 1.8.9 and 1.9.4), that has patched a vulnerability in configmaps and now they can't be mounted as volumes for write. See issue: https://github.com/kubernetes/kubernetes/issues/60814
The previous version with "curl" would break as well in newest Kubernetes, because the path /var/lib/grafana/import is a configmap mounted as a volume.

~~So I guess we should get rid of "dashboards-import-volume" and the problem will be fixed. This volume wasn't ever needed really.~~
**Edit**: the volume is needed, but has to be used as read-only. The original idea, to have another directory for downloading dashboards seems correct :)

The problem will also go away when either https://github.com/kubernetes/charts/pull/3514 or https://github.com/kubernetes/charts/pull/4713 gets merged so that we can use Grafana's 5 built-in dashboard provisioning.

**Edit**: my idea to remove the configmap volume wasn't correct, I have edited my comment.

**Comment by Jokero** [comment]
@jgoclawski Thanks for the detailed comment. Now it's clear :)

**Comment by rtluckie** [comment]
@hmacias this is fixed with https://github.com/kubernetes/charts/pull/4713

**Comment by hmacias** [comment]
@rtluckie 

I will close this one out once #4713 is merged. That way I can get the latest updates and try the new method to download/install dashboards.

Thanks.

**Comment by Jokero** [comment]
Can confirm last chart version works

**Comment by hmacias** [comment]
@rtluckie 

I confirm I was able to install Grafana dashboards successfully setting up a default Prometheus datasource and a dashboard import config like the following:

```
dashboards:
 kubernetes-monitoring:
   gnetId: 315
   revision: 3
   datasource: prometheus-backend-stats
```

The changes in this pull request are no longer needed nor valid given the changes introduced by #4713. Closing this issue now.

Thank you @rtluckie!
